src/ucs/sys/topo/base/topo.c

```diff
@@ -32,6 +33,8 @@
 /*
  * Function pointer used to refer to specific implementations of
  * ucs_topo_get_memory_distance function by topology modules.
+ * This function calculates the memory distance of the device
+ * based on the machine topology and process affinity.
```

Comment:
This function estimates the distance between the device and the system memory used by the current thread according to its CPU affinity.